api/api.go

```diff
@@ -50,53 +52,81 @@ func configureExtensionsApi(config *core.Config, router *mux.Router, ctx context
 }
 
 func (api *ExtensionsApi) extensionsHandler(rw http.ResponseWriter, r *http.Request) {
+	if websocket.IsWebSocketUpgrade(r) {
+		upgrader := websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+			CheckOrigin: func(r *http.Request) bool {
+				return true
+			},
+		}
+
+		websocket, err := upgrader.Upgrade(rw, r, nil)
+		if err != nil {
+			return
+		}
+
+		notifications := make(chan StatusUpdate)
+
+		notify := func(notification StatusUpdate) {
+			notifications <- notification
+		}
+
+		go func() {
+			for notification := range notifications {
+				encoder := json.NewEncoder(rw)
+				encoder.Encode(extensionsResponse{api.Extensions, api.Version})
+				websocket.WriteJSON(&notification)
+			}
+		}()
+
+		api.registerClient(websocket, notify)
+
+		websocket.SetCloseHandler(func(code int, text string) error {
+			close(notifications)
+			api.unregisterClient(websocket)
+			return nil
+		})
+
+		websocket.WriteJSON(StatusUpdate{Type: "Connected", Extensions: api.Extensions})
+	}
+
+	api.extensionsJSONHandler(rw, r)
```

Comment:
I think this has to go in an else branch. This might cause the warnings on the console:

```sh
2021/09/03 16:19:11 http: response.Write on hijacked connection from encoding/json.(*Encoder).Encode (stream.go:231)
```

Reply:
Sorry, forgot to push